cg/cli/workflow/raredisease/base.py

```diff
@@ -41,6 +64,62 @@ def config_case(context: CGConfig, case_id: str, dry_run: bool) -> None:
         raise click.Abort() from error
 
 
+@raredisease.command("run")
+@ARGUMENT_CASE_ID
+@OPTION_LOG
+@OPTION_WORKDIR
+@OPTION_FROM_START
+@OPTION_PROFILE
+@OPTION_CONFIG
+@OPTION_PARAMS_FILE
+@OPTION_REVISION
+@OPTION_COMPUTE_ENV
+@OPTION_USE_NEXTFLOW
+@OPTION_TOWER_RUN_ID
+@DRY_RUN
+@click.pass_obj
+def run(
+    context: CGConfig,
+    case_id: str,
+    log: str,
+    work_dir: str,
+    from_start: bool,
+    profile: str,
+    config: str,
+    params_file: str,
+    revision: str,
+    compute_env: str,
+    use_nextflow: bool,
+    nf_tower_id: str | None,
+    dry_run: bool,
+) -> None:
+    """Run raredisease analysis for given CASE ID."""
+    analysis_api: RarediseaseAnalysisAPI = context.meta_apis[MetaApis.ANALYSIS_API]
+    analysis_api.status_db.verify_case_exists(case_internal_id=case_id)
+
+    command_args: NfCommandArgs = NfCommandArgs(
+        **{
+            "log": analysis_api.get_log_path(
+                case_id=case_id,
+                workflow=analysis_api.workflow,
+                log=log,
+            ),
+            "work_dir": analysis_api.get_workdir_path(case_id=case_id, work_dir=work_dir),
+            "resume": not from_start,
+            "profile": analysis_api.get_profile(profile=profile),
+            "config": analysis_api.get_params_file_path(case_id=case_id, params_file=params_file),
+            "name": case_id,
+            "compute_env": compute_env or analysis_api.get_compute_env(case_id=case_id),
+            "revision": revision or analysis_api.revision,
+            "wait": NfTowerStatus.SUBMITTED,
+            "id": nf_tower_id,
+        }
+    )
```

Comment:
This logic can likely be a function within the analysis api and does not need to be part of the CLI layer. 

We can either adjust the `run_nextflow_analysis` to retrieve the command args.

or add a function to the analysisapi called something along the line of 
`get_command_args` that returns the command args that are built here.

This way we can keep our CLI layer cleaner and more easily understood.

Reply:
At the moment, raredisease is building the command arguments differently than taxprofiler and rnafusion: the config file is used instead of the parameter files. I was going to wait with the migration to run_nextflow_analysis until we unify all nextflow pipeline (after the extensive/validation test on raredisease)

Reply:
Is that ok, or should I try to it now?

Reply:
It can be part of `RarediseaseAnalysisAPI`
Within the `RarediseaseAnalysisAPI`
1. Move `verify_case_exists` into `run_nextflow_analysis` with error handling
2. create a new function `get_command_args()`
3. Call `get_command_args()` in `run_nexflow_analysis`

Then the CLI will me much cleaner and easier to test. 

From what I understand is that each workflow have their own "analysisapi" so it can be handled within there even if it differs. 

_I was going to wait with the migration to run_nextflow_analysis until we unify all nextflow pipeline (after the extensive/validation test on raredisease)_

I don't understand this. this PR adds the function below?

Reply:
Ah, I perhaps see now. Is it due to how the `run`cli is called somewhere else? If that clashes too much then resolve for now. We can likely update that later then

Reply:
I do like this suggestion. Then we could possibly move it as a general nf_analysis command and just import it

Reply:
Addressed, using a general nf_analysis function for "run"